Fix issue #5803: Document repo-specific prompt customization

openhands-agent opened the pull request:
This pull request fixes #5803.

The issue has been successfully resolved as the AI agent has created comprehensive documentation for both repo-specific prompt customization features (.openhands_instructions file and .openhands directory) exactly as requested. The documentation was placed in the correct location (/workspace/docs/modules/usage/prompting/repo-specific-customization.md) under the "Prompting" section as specified in the original issue.

PR Review Message:
This PR adds comprehensive documentation for OpenHands' repo-specific customization features. The new documentation:
- Explains both the `.openhands_instructions` file and `.openhands` directory features
- Provides detailed usage instructions and examples for each feature
- Includes best practices for implementation
- Is properly placed in the documentation hierarchy under Prompting
- Fills the documentation gap identified in the original issue

The documentation should help users better understand and utilize these previously undocumented features. Please review the content for accuracy and completeness.

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:2dd7a71-nikolaik   --name openhands-app-2dd7a71   docker.all-hands.dev/all-hands-ai/openhands:2dd7a71
```

neubig commented on docs/modules/usage/prompting/repo-specific-customization.md:
```diff
@@ -0,0 +1,140 @@
+
+### What to Include
+
+Your `.openhands_instructions` file should contain:
```
```suggestion
Your `.openhands_instructions` file can contain information like the following:
```

openhands-agent replied:
Openhands fix success summary


The feedback appears to be suggesting a specific formatting change to introduce code block syntax (using triple backticks) for the `.openhands_instructions` file example. However, the AI's response doesn't directly address this formatting suggestion - instead it talks about updating the content of the example and simplifying other sections.

While the AI may have made other improvements to the documentation, it hasn't specifically implemented the requested formatting change to wrap the `.openhands_instructions` file content in code blocks. The feedback was very specific about syntax (```suggestion) which indicates this was meant to be a formatting change, not a content change.

To properly resolve this feedback, the AI should add the triple backtick syntax around the `.openhands_instructions` file example in the documentation.

neubig commented on docs/modules/usage/prompting/repo-specific-customization.md:
```diff
@@ -0,0 +1,140 @@
+
+## The `.openhands` Directory
+
+The `.openhands` directory allows you to create repository-specific micro-agents that extend OpenHands' capabilities for your project. These micro-agents are defined in markdown files within this directory.
```
Add a link to the documentation page about micro-agents.
You do not need to include any information that is already included in that documentation page, instead just refer to that page.

openhands-agent replied:
Openhands fix success summary


The feedback has been successfully incorporated. The AI agent confirmed they simplified the `.openhands` directory section by removing redundant information and adding a link to the micro-agents documentation page, which was exactly what was requested in the feedback. This avoids duplication of information while ensuring users can still access the complete details about micro-agents through the linked documentation.

The response can be sent to the reviewer as:
"The `.openhands` directory section has been updated to include a link to the micro-agents documentation page, removing redundant information and ensuring users have access to the complete, up-to-date documentation about micro-agents in a single location."

openhands-agent replied:
Openhands fix success summary


Based on the feedback and the AI's last message, it appears the feedback has not been fully addressed. The feedback specifically requested to add a link to the micro-agents documentation page and remove redundant information, but the AI's response only mentions:
1. Adding the file to the sidebar
2. Updating code block formatting

There is no mention of adding the requested link to the micro-agents documentation page or removing redundant information. The AI's response seems to be addressing different changes than what was requested in the feedback.

To resolve this, the AI should explicitly confirm they have:
1. Added a link to the micro-agents documentation page
2. Removed any redundant information that's already covered in the micro-agents documentation
3. Verified these specific changes in the repo-specific-customization.md file

openhands-agent commented:
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the pr! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12497560128).

openhands-agent commented:
Here's a concise overview of the changes and remaining issues:

RESOLVED:
- Successfully simplified the `.openhands` directory section
- Added appropriate link to micro-agents documentation
- Removed redundant information

STILL PENDING:
- Need to implement the requested formatting change by adding triple backticks (```) around the `.openhands_instructions` file example

RECOMMENDATION:
Complete the documentation update by implementing the missing code block formatting for the `.openhands_instructions` file example.

neubig reviewed (commented):
Please check if `repo-specific-customization.md` is added to the sidebar within the "Prompting" directory below the "Micro-agents" section.

openhands-agent commented:
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the pr! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12497620732).

openhands-agent commented:
Overview of Changes:

✗ INCOMPLETE - Issues not fully resolved

Changes Made:
- Added file to sidebar
- Updated code block formatting

Still Pending:
1. Adding link to micro-agents documentation page
2. Removing redundant information
3. Verification of changes in repo-specific-customization.md

Recommendation:
Complete the remaining requested changes to fully address the original feedback.

neubig commented:
Fixing this issue with OpenHands seems to be blocked on resource constraints when building the site? When I run it in the online app it crashes as unresponsive. This may be fixed by being able to use larger machines.

neubig commented:
I'm going to take another stab at this later, I think this doc is already out of date.
